### ISSUE TITLE
Issue with actcache: actId collision

### ISSUE THREAD

**Issue by dhautotlf** [senecajs/seneca#503]
Hi,

We have 4 hapijs processes running in parallel + 4 seneca client instances all behind 2 loadbalancers.
We have `actid` collision when running in production.

Flow:
1 - Calling action1 gives `OUT 69/l8 cmd:*,role:a` on api_node_01 which calls **cmd:a,role:a** on srv_node_01
Some days later:
2 - Calling action2 gives `IN 69/l8 cmd:*,role:a` on api_node_02 which calls **cmd:b,role:a** on srv_node_01

Which returns the first call result which corresponds to **another action result**.

Is {actcache: {active: false}} the only solution to avoid this issue or am I doing something wrong here? 

Thanks :)

**Comment by mcdonnelldean** [comment]
@dhautotlf Turning actcache off will certainly work but those log statements look odd.

Do you have a pin or act that matches that log? Specifically `cmd:*,role:a`.

**Comment by dhautotlf** [comment]
@mcdonnelldean yes turning actcache off do the trick. 

Yes we have a act that matches this in our srv node : ` seneca.add({role: 'a', cmd: 'a'}, handle1);` and  seneca.add({role: 'a', cmd: 'b'}, handle2);`

**Comment by mcdonnelldean** [comment]
@dhautotlf Do these then back on to listens or clients __specifically__ with this pin, `{pin: 'role:a'}` or `{pin: 'role:a,cmd:*'}` (or their json equivs) ?

**Comment by dhautotlf** [comment]
@mcdonnelldean yes
On api side : 
`si.client({
        ...
        pins: [{role: 'a', cmd: '*'}],
        ...
    })`
Then :
`req.seneca.act('role:a,cmd:a', ...`
`req.seneca.act('role:a,cmd:b', ...`

On srv side :
`seneca.add({role: a, cmd: 'a'}, handle1);`
`seneca.add({role: a, cmd: 'b'}, handle2);`

**Comment by mcdonnelldean** [comment]
Awesome. The reason I ask is, I think the problem is actCache caching the pin, not the match itself. This would explain the issue you are having. We are flat out for 3.0 right now but lets keep this open so we can look at it once the release drops as a point fix.

**Comment by dhautotlf** [comment]
Fantastic. 👍 
Yes I was thinking so too. I will keep it disabled for now.
Ow and It would be nice to have a reference of this caching system in the doc somewhere :) 
Thanks for your help !

**Comment by mcdonnelldean** [comment]
@dhautotlf Agreed! It's on the list. It is mentioned casually in a couple of places but I think it needs a solid right up. Or at the very least an example explaining it in `docs/examples`

**Comment by rjrodger** [comment]
@dhautotlf your ids are in the format XX/YY - this the debug form - do you have the `debug.short_logs` option enabled? The ids should be much longer (to prevent collisions).

**Comment by mcdonnelldean** [comment]
@dhautotlf To note, by default we are going to take this out in the next version. See #509 

I'll close for now. If you have any more questions if you want to answer the above, please do reopen. 👍

**Comment by dhautotlf** [comment]
@rjrodger indeed we kept `debug.short_logs` option enabled. I though it would just impact the logging.
Isn't it weird that this logging option changes the cache behaviour ?

**Comment by mcdonnelldean** [comment]
@dhautotlf It's an oversight I think. It's something that needs to be fixed for sure anyway.

Good news though, this will be switched off by default in 3.0. It's landed already in master. We'll most likely devolve it into a plugin and fix it up as something you can optionally add. 

Couple of hours until release.
